### PR TITLE
[Merged by Bors] - chore: forward-port leanprover-community/mathlib#18980

### PR DESCRIPTION
**Pull request by eric-wieser**
---

- [x] depends on: #3888

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#3888
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Data/Real/ENNReal.lean [hunk 1]
```diff
@@ -4,7 +4,7 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl, Yury Kudryashov
 
 ! This file was ported from Lean 3 source module data.real.ennreal
-! leanprover-community/mathlib commit 29cb56a7b35f72758b05a30490e1f10bd62c35c1
+! leanprover-community/mathlib commit ec4b2eeb50364487f80421c0b4c41328a611f30d
```
Comment (eric-wieser) [inline Mathlib/Data/Real/ENNReal.lean:7]: Note that this change skips over https://github.com/leanprover-community/mathlib4/pull/3888; so we need to merge that PR first.

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#3888~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Topology/Instances/ENNReal.lean [hunk 2]
```diff
@@ -899,6 +909,10 @@ protected theorem tsum_mul_right : (∑' i, f i * a) = (∑' i, f i) * a := by
   simp [mul_comm, ENNReal.tsum_mul_left]
 #align ennreal.tsum_mul_right ENNReal.tsum_mul_right
 
+protected theorem tsum_const_smul {R} [SMul R ℝ≥0∞] [IsScalarTower R ℝ≥0∞ ℝ≥0∞] (a : R) :
+    (∑' i, a • f i) = a • ∑' i, f i := by
+  simpa only [smul_one_mul] using @ENNReal.tsum_mul_left _ (a • (1 : ℝ≥0∞)) _
+#align ennreal.tsum_const_smul ENNReal.tsum_const_smul
```
Comment (eric-wieser) [inline Mathlib/Topology/Instances/ENNReal.lean:915]:
```suggestion
#align ennreal.tsum_const_smul ENNReal.tsum_const_smul

```

**Comment by ADedecker** [comment]
I'm not sure how to check that all SHAs are as they should be, otherwise LGTM. 

maintainer merge

**Comment by eric-wieser** [comment]
> I'm not sure how to check that all SHAs are as they should be, otherwise LGTM.

If you look at the file diff view, a bot leaves an annotation with a link to check for each file

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**